src/treetensornetworks/projttns/projttn_apply.jl

```diff
@@ -71,3 +71,28 @@ function make_environment(P::ProjTTNApply, state::AbstractTTN, e::AbstractEdge)
   )
   return P
 end
+
+function projected_operator_tensors(P::ProjTTNApply)
+  environments = ITensor[environment(P, edge) for edge in incident_edges(P)]
+  # manual heuristic for contraction order fixing: for each site in ProjTTN, apply up to
+  # two environments, then TTN tensor, then other environments
+  itensor_map = Union{ITensor,OneITensor}[] # TODO: will a Hamiltonian TTN tensor ever be a OneITensor?
+  for j in sites(P)
+    push!(itensor_map, init_state(P)[j])
+  end
+  if on_edge(P)
+    append!(itensor_map, environments)
+  else
+    for s in sites(P)
+      site_envs = filter(hascommoninds(operator(P)[s]), environments)
+      frst, scnd, rst = _separate_first_two(site_envs)
+      site_tensors = vcat(frst, scnd, operator(P)[s], rst)
+      append!(itensor_map, site_tensors)
+    end
+  end
+  return itensor_map
+end
+
+function product(P::ProjTTNApply, v::ITensor)::ITensor
+  return noprime(contract(P, v))
+end
```

Comment:
Can we delete this in favor of the generic `AbstractProjTTN` version?

Reply:
yes